### PR TITLE
CDP connectors - manage enums per OpenApi spec

### PR DESCRIPTION
**Pull request by LucGenetier**
Add CDP compatibility flag
When enabled, we don't require format=enum

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs [hunk 1]
```diff
@@ -64,6 +64,11 @@ public enum ConnectorCompatibility
         // Swagger File Conformity
         // - parameters appear in the order specified in the swagger file
         // - internal visible parameters are completely hidden (required/optional, with or without default value)
-        SwaggerCompatibility = 2
+        SwaggerCompatibility = 2,
+
+        // Swagger File Conformity for CDP connectors
+        // - same as Swagger File Conformity
+        // - does not require format="enum" for identifying enumerations
+        CdpCompatibility = 3
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs:72]: >CdpCompatibility = 3 [](http://example.com/codeflow?start=8&length=20)

can you add test? 

1 swagger, applying each flag

File: src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs [hunk 2]
```diff
@@ -1448,7 +1449,8 @@ private ConnectorParameterInternals Initialize()
 
                                             bodyPropertyHiddenRequired = ConnectorSettings.Compatibility != ConnectorCompatibility.PowerAppsCompatibility || !requestBody.Required;
                                         }
-                                        else if (ConnectorSettings.Compatibility == ConnectorCompatibility.SwaggerCompatibility)
+                                        else if (ConnectorSettings.Compatibility == ConnectorCompatibility.SwaggerCompatibility || 
+                                                 ConnectorSettings.Compatibility == ConnectorCompatibility.CdpCompatibility)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs:1453]: Can we use an extension method instead? 

So we don't need each caller to make this decision. 



ConnectorSettings.Compatibility.IsXXXX()

and 

public static IsXXXX(this ConnectorCompatibility settings)
{
return settings == ConnectorCompatibility.SwaggerCompatibility ||      
                                            settings == ConnectorCompatibility.CdpCompatibility);

}

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs [hunk 3]
```diff
@@ -431,26 +431,27 @@ internal static ConnectorType GetConnectorType(this ISwaggerParameter openApiPar
...
-                            else
-                            {
-                                return new ConnectorType(error: $"Unsupported enum type {schema.Enum.GetType().Name}");
-                            }
+                            return new ConnectorType(schema, openApiParameter, FormulaType.Blob);
+                    }
 
-                        default:
-                            return new ConnectorType(schema, openApiParameter, FormulaType.String);
+                    if (schema.Enum != null && (settings.Compatibility == ConnectorCompatibility.CdpCompatibility || schema.Format == "enum"))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:437]: > (settings.Compatibility == ConnectorCompatibility.CdpCompatibility || schema.Format == "enum") [](http://example.com/codeflow?start=46&length=95)

similar, can you move this to extension method? 

The idea is that all the policy around what the settings mean is then captured in a central place.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs (hunk 1 above)
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs:72]: >CdpCompatibility [](http://example.com/codeflow?start=8&length=16)

This really is a misnomer.  It's what we should have originally done for SwaggerCompat,  (but can't change swagger compat due to breaking changes). 
But not sure what a better name would be.

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs (hunk 1 above)
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Connectors/Public/ConnectorSettings.cs:72]: Not sure what better name to use

**Pull request merged**